src/crest/params_encoders/enumerated_flat_params_encoder.cr

```diff
@@ -1,5 +1,10 @@
 module Crest
   class EnumeratedFlatParamsEncoder < Crest::ParamsEncoder
+    # ```
+    # Encoded array params start index, default 1
+    # ```
+    class_property array_start_index = 1
```

Comment:
I think it's better to have a new encoder class with array index starting with 0.